src/Core/Grid/Definition/Factory/CartRuleGridDefinitionFactory.php

```diff
@@ -0,0 +1,262 @@
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\BulkActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\DateTimeColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ToggleColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\DataColumn;
+use PrestaShop\PrestaShop\Core\Grid\Filter\Filter;
+use PrestaShop\PrestaShop\Core\Grid\Filter\FilterCollection;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShopBundle\Form\Admin\Type\DateRangeType;
+use PrestaShopBundle\Form\Admin\Type\SearchAndResetType;
+use PrestaShopBundle\Form\Admin\Type\YesAndNoChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+
+/**
+ * Class responsible for providing columns, filters, actions for cart price rule list.
+ */
+final class CartRuleGridDefinitionFactory extends AbstractGridDefinitionFactory
+{
+    const GRID_ID = 'cart_rule';
+
+    public function __construct(
+        HookDispatcherInterface $hookDispatcher
+    ) {
+        parent::__construct($hookDispatcher);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId(): string
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName(): string
+    {
+        return $this->trans('Cart rules', [], 'Admin.Catalog.Feature');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns(): ColumnCollection
+    {
+        return (new ColumnCollection())
+            ->add((new BulkActionColumn('bulk'))
+            ->setOptions([
+                'bulk_field' => 'id_cart_rule',
+            ])
+            )
+            ->add((new DataColumn('id_cart_rule'))
+            ->setName($this->trans('ID', [], 'Admin.Global'))
+            ->setOptions([
+                'field' => 'id_cart_rule',
+            ])
+            )
+            ->add((new DataColumn('name'))
+            ->setName($this->trans('Name', [], 'Admin.Global'))
+            ->setOptions([
+                'field' => 'name',
+            ])
+            )
+            ->add((new DataColumn('priority'))
+            ->setName($this->trans('Priority', [], 'Admin.Global'))
+            ->setOptions([
+                'field' => 'priority',
+            ])
+            )
+            ->add((new DataColumn('code'))
+            ->setName($this->trans('Code', [], 'Admin.Global'))
+            ->setOptions([
+                'field' => 'code',
+            ])
+            )
+            ->add((new DataColumn('quantity'))
+            ->setName($this->trans('Quantity', [], 'Admin.Catalog.Feature'))
+            ->setOptions([
+                'field' => 'quantity',
+            ])
+            )
+            ->add((new DateTimeColumn('date_to'))
+            ->setName($this->trans('Expiration date', [], 'Admin.Catalog.Feature'))
+            ->setOptions([
+                'format' => 'Y-m-d H:i',
```

Comment:
maybe store format in some settings obj? or isnt it necessary?
like Domain/CartRule/CartRuleSettings::EXPIRATION_DATE_FORMAT ?
@matks, wdyt?

Reply:
I see that order controller is using @=service("prestashop.adapter.legacy.context").getContext().language.date_format_full
For this. Maybe I should use same thing here?

Reply:
Yes, please use `@=service("prestashop.adapter.legacy.context").getContext().language.date_format_full`

It returns a format that is usually used by people using this language and locale. It adapts format to user, according to his country/locale preferences.